File: Storage/Java/app/src/main/java/com/naufalprakoso/javaimagegallery/ImageAdapter.java
package com.naufalprakoso.javaimagegallery;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ViewHolder> {

    private Context context;
    private ArrayList<Image> images;

    ImageAdapter(Context context, ArrayList<Image> images) {
        this.context = context;
        this.images = images;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_image, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        viewHolder.txtName.setText(images.get(i).getName());
        Glide.with(context).load(images.get(i).getImage()).into(viewHolder.imageView);
    }

    @Override
    public int getItemCount() {
        return images.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imageView;
        TextView txtName;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.img_data);
            txtName = itemView.findViewById(R.id.txt_img_name);
        }
    }
}

File: Storage/Kotlin/app/src/main/java/com/naufalprakoso/kotlinfirebasestorage/ImageAdapter.kt
package com.naufalprakoso.kotlinfirebasestorage

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_image.view.*

class ImageAdapter(
	private val context: Context,
	private val images: ArrayList<Image>
) : RecyclerView.Adapter<ImageAdapter.ViewHolder>() {
	
	override fun onCreateViewHolder(p0: ViewGroup, p1: Int): ImageAdapter.ViewHolder =
		ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_image, p0, false))
	
	override fun getItemCount(): Int = images.size
	
	override fun onBindViewHolder(p0: ImageAdapter.ViewHolder, p1: Int) =
			p0.bindItem(images[p1])
	
	class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
		fun bindItem(image: Image){
			itemView.txt_img_name.text = image.name
			Glide.with(itemView.context).load(image.image).into(itemView.img_data)
		}
	}
}